src/Catrobat/Controller/Admin/DownloadLogController.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace App\Catrobat\Controller\Admin;
+
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\BinaryFileResponse;
+use Symfony\Component\HttpFoundation\File\File;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpFoundation\ResponseHeaderBag;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use Symfony\Component\Routing\Annotation\Route;
+
+class DownloadLogController extends AbstractController
+{
+  /**
+   * @Route("/downloadLogs/", name="log_download")
+   */
+  public function downloadLogAction(Request $request = null): Response
```

Comment:
That's way too powerful.
I should have stated it better the first time Sorry for that.

However, with this approach users can download everything from our server.
E.g.:
![image](https://user-images.githubusercontent.com/40868718/103529399-e4224480-4e85-11eb-90c4-66f94bd9930b.png)

It is important that no '..' access is possible. The easiest way might be to zip the whole /var/log folder and download this zipped folder. Completely independent of any user parameters.

Secondly, only the admin must be allowed to download to log files. So there is some authentication necessary. I guess we should just comment this route out and create a new ticket and do it within CAPI. Because the rest of your ticket is fine and looks good! However, you can also do in this PR since this ticket has a low priority to be merged.

Your opinions?
@HCrane @mahmoudZkhalil

Reply:
since this is an admin page, i didn't think so much about securing the download but i came up with a simple solution which is to limit the download for only this folder.